libs/sdk-common/src/input_parser.rs

```diff
@@ -186,6 +195,19 @@ pub async fn parse(
 ) -> Result<InputType> {
     let input = input.trim();
 
+    let mut dns_resolvers_opts = ResolverOpts::default();
+    dns_resolvers_opts.validate = true;
```

Comment:
I'm getting a line in the log:
`validate option is only available with 'dnssec' feature`

Tip:
Add the following to cargo.toml:
```
hickory-proto = { version = "0.24.2", features = ["dnssec"] }
hickory-resolver = { version = "0.24.2", features = ["dnssec"] }
```

However, when I try that on some BIP-353 domains I get: `proto error: self-signed dnskey is invalid`

Perhaps this is useful? https://github.com/TheBlueMatt/dnssec-prover

Reply:
I am not able to reproduce your log, but reading inside the documentation I found this message `To enable DNSSEC, one of the features dnssec-openssl or dnssec-ring must be enabled.` (https://crates.io/crates/hickory-resolver#:~:text=To%20enable%20DNSSEC%2C%20one%20of%20the%20features%20dnssec%2Dopenssl%20or%20dnssec%2Dring%20must%20be%20enabled.)

I tried to do that. Can you explain me better where did you find this log?

Reply:
The log `proto error: self-signed dnskey is invalid` I no longer get with `dnssec-ring`.

The log `validate option is only available with 'dnssec' feature` is also no longer there with `dnssec-ring`.

I think that resolves it!